src/Microsoft.Azure.SignalR.Management/ServiceManagerOptions.cs

```diff
@@ -36,40 +34,73 @@ public class ServiceManagerOptions
         /// <summary>
         /// Gets or sets the service endpoint for accessing Azure SignalR Service.
         /// </summary>
-        public ServiceEndpoint ServiceEndpoint
+        public ServiceEndpoint ServiceEndpoint { get; set; }
+
+        /// <summary>
+        /// Gets or sets the service endpoints for accessing Azure SignalR Service and switches to multi-endpoint mode.
+        /// </summary>
+        internal ServiceEndpoint[] ServiceEndpoints { get; set; } //not ready for public use
+
+        /// <summary>
+        /// Gets or sets the transport type to Azure SignalR Service. Default value is Transient.
+        /// </summary>
+        public ServiceTransportType ServiceTransportType { get; set; } = ServiceTransportType.Transient;
+
+        /// <summary>
+        /// A unified way to get the service endpoint(s) set by "ConnectionString", "ServiceEndpoint" or "ServiceEndpoints".
+        /// </summary>
+        internal ServiceEndpoint[] UnifiedEndpoints
         {
             get
             {
-                if (_serviceEndpoint == null)
+                ValidateOptions();
+                if (ConnectionString != null)
                 {
-                    _serviceEndpoint = new ServiceEndpoint(ConnectionString, EndpointType.Secondary);
+                    return new ServiceEndpoint[] { new ServiceEndpoint(ConnectionString) };
+                }
+                if (ServiceEndpoint != null)
+                {
+                    return new ServiceEndpoint[] { ServiceEndpoint };
+                }
+                else
+                {
+                    return ServiceEndpoints;
                 }
-                return _serviceEndpoint;
-            }
-            set
-            {
-                _serviceEndpoint = value;
             }
         }
 
-        /// <summary>
-        /// Gets or sets the transport type to Azure SignalR Service. Default value is Transient.
-        /// </summary>
-        public ServiceTransportType ServiceTransportType { get; set; } = ServiceTransportType.Transient;
+        internal string ProductInfo { get; set; }
 
         internal void ValidateOptions()
         {
-            if (_serviceEndpoint == null)
-            {
-                ValidateConnectionString();
-            }
+            ValidateServiceEndpoint();
             ValidateServiceTransportType();
         }
 
-        private void ValidateConnectionString()
+        private void ValidateServiceEndpoint()
         {
-            // if the connection string is invalid, exceptions will be thrown.
-            ConnectionStringParser.Parse(ConnectionString);
+            var notNullCount = 0;
+            if (ConnectionString != null)
+            {
+                notNullCount += 1;
+            }
+            if (ServiceEndpoint != null)
+            {
+                notNullCount += 1;
+            }
+            if (ServiceEndpoints != null)
```

Comment:
what if `ServiceEndpoints ` is set to an empty array

Reply:
A check in the setter added.